pkg/controller/process.go

```diff
@@ -43,6 +43,15 @@ func (c AppGwIngressController) Process(event events.Event) error {
 		EnvVariables:         environment.GetEnv(),
 	}
 
+	// Mutate the list of Ingresses by removing ones that AGIC should not be creating configuration.
+	if cbCtx.EnvVariables.EnableBrownfieldDeployment == "true" {
+		for idx, ingress := range cbCtx.IngressList {
+			glog.V(5).Infof("Original Ingress[%d] Rules: %+v", idx, ingress.Spec.Rules)
+			cbCtx.IngressList[0].Spec.Rules = brownfield.PruneIngressRules(ingress, cbCtx.ProhibitedTargets, cbCtx.ManagedTargets)
```

Comment:
is there any issue in case `len(IngressRule) ==  0` in config generation ?

Reply:
That should be `idx` here.
If the list of ingresses is empty - we won't ever get in this loop here.